lib/src/components/NxNumberInput/stateful/NxStatefulNumberInput.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef, useState } from 'react';
+
+import { initialState, userInput } from '../../NxTextInput/stateHelpers';
+import NxNumberInput from '../NxNumberInput';
+
+import { Props, propTypes } from './types';
+export { Props, propTypes } from './types';
+
+/**
+ * Standard text input with pristine state tracking and pluggable validation handling
+ * @param defaultValue The initial value rendered in the text input
+ * @param validator A function that validates user-inputted changes to the text field value. Accepts the new value
+     as a string and returns zero or more validation error messages
+ * @param onChange A callback for when the user changes the value of the text box (e.g. by typing a letter)
+ * @param onKeyPress A callback for when the user presses a key that not necessarily changes the value of the text box
+ * See the doc page for this component for information about other supported attributes.
+ */
+const NxStatefulNumberInput = forwardRef<HTMLDivElement, Props>(
+    function NxStatefulNumberInput(props, ref) {
+      const { defaultValue, onChange, validator, ...attrs } = props,
+          [state, setState] = useState(initialState(defaultValue || '', validator));
+
+      function changeHandler(newValue: string) {
+        setState(userInput(validator || null, newValue));
+
+        if (onChange) {
+          onChange(newValue);
+        }
+      }
+
+      return <NxNumberInput type='number'
```

Comment:
This `type` doesn't do anything here since it's already hard-coded in the `NxNumberInput`, right?

Reply:
> This `type` doesn't do anything here since it's already hard-coded in the `NxNumberInput`, right?

No it shouldn't be there: https://github.com/sonatype/sonatype-react-shared-components/pull/486/commits/a30cce6c4eba2979a2669e8efdf08641bc4e9527